Ballerina-Training/Ballerina-Forum-Application/back-end/bal-forum/endpoints.bal

```diff
@@ -0,0 +1,117 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/task;
+import ballerinax/googleapis.gmail;
+import ballerinax/mysql;
+import ballerinax/mysql.driver as _;
+import ballerinax/nats;
+
+# Database related configurations
+type DatabaseConfig record {|
+    # The database host
+    string host;
+    # The database user
+    string user;
+    # The database user password
+    string password;
+    # The database name
+    string database;
+    # The database port
+    int port;
+|};
+
+configurable DatabaseConfig dbConfig = ?;
+
+final mysql:Client forumDbClient = check new (...dbConfig);
+
+# Sentiment endpoint related configurations
+type SentimentEndpointConfig record {|
+    # The endpoint URL
+    string endpointUrl;
+    # The retry interval
+    decimal retryInterval;
+    # The authentication configuration
+    record {|
+        # The refresh URL
+        string refreshUrl;
+        # The client ID
+        string clientId;
+        # The client secret
+        string clientSecret;
+        # The refresh token
+        string refreshToken;
+    |} authConfig;
+|};
+
+configurable SentimentEndpointConfig sentimentEpConfig = ?;
+
+final http:Client sentimentClient = check new (sentimentEpConfig.endpointUrl,
+    config = {
+        retryConfig: {
+            interval: sentimentEpConfig.retryInterval
+        },
+        auth: {
+            ...sentimentEpConfig.authConfig,
+            clientConfig: {
+                secureSocket: {
+                    cert: "./resources/public.crt"
+                }
+            }
+        },
+        secureSocket: {
+            cert: "./resources/public.crt"
+        }
+    }
+);
+
+final nats:Client natsClient = check new (url = nats:DEFAULT_URL);
+
+# Gmail related configurations
+type GmailAuthConfig record {|
```

Comment:
Can't we reuse `OAuth2RefreshTokenGrantConfig`, which is used in the gmail module? We can still provide just these 4, right?

Reply:
The one in the gmail module is coming from a different module called `ballerinax/'client.config`. I think that is a module used for internal purpose to overwrite the client configurations in the connector. IMO it is better to not exposing it here